arch/arm/src/armv7-m/Toolchain.defs

```diff
@@ -112,6 +108,16 @@ else # ifeq ($(CONFIG_ARCH_CORTEXM3),y)
   TOOLCHAIN_MFLOAT   := -mfloat-abi=soft
 endif
 
+ifeq ($(CONFIG_ARCH_FPU),y)
+  ifeq ($(CONFIG_ARM_FPU_ABI_SOFT),y)
+    TOOLCHAIN_MFLOAT += -mfloat-abi=softfp
+  else
+    TOOLCHAIN_MFLOAT += -mfloat-abi=hard
+  endif
+else
+  TOOLCHAIN_MFLOAT = -mfloat-abi=soft
```

Comment:
```suggestion
  TOOLCHAIN_MFLOAT += -mfloat-abi=soft
```

Reply:
Or remove from `else # ifeq ($(CONFIG_ARCH_CORTEXM3),y)` case

Reply:
Or set as else for `ifeq ($(CONFIG_ARCH_FPU),y)` in above `ifeq ($(CONFIG_ARCH` selection